securedrop/journalist.py

```diff
@@ -242,9 +242,19 @@ def admin_reset_two_factor_hotp():
     otp_secret = request.form.get('otp_secret', None)
     if otp_secret:
         user = Journalist.query.get(uid)
-        user.set_hotp_secret(otp_secret)
-        db_session.commit()
-        return redirect(url_for('admin_new_user_two_factor', uid=uid))
+        try:
+            user.set_hotp_secret(otp_secret)
+        except TypeError as e:
+            if "Non-hexadecimal digit found" in str(e):
+                flash("Invalid secret format: non-hexadecimal digit found")
+            elif "Odd-length string" not in str(e)):
+                flash("Invalid secret format: odd-length string")
```

Comment:
The word "string" may not be clear to many end users, so we could say "odd-length secret" instead. Adding a statement like "Did you mistype the secret?" to prompt the user may help them understand what the implications of an odd-length string are.